catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/WorkspaceTransformer.java

```diff
@@ -277,4 +292,38 @@ public Metacard toMetacardFromXml(Serializable xml) {
 
     return null;
   }
+
+  public void addListActions(Metacard workspaceMetacard, Map<String, Object> workspaceAsMap) {
+    final List<Map<String, Object>> listActions = getListActions(workspaceMetacard);
+    final List<Map<String, Object>> lists =
+        (List<Map<String, Object>>) workspaceAsMap.get(WorkspaceAttributes.WORKSPACE_LISTS);
+    if (lists != null) {
+      lists.forEach(list -> list.put(ACTIONS_KEY, listActions));
+    }
+  }
+
+  private List<Map<String, Object>> getListActions(Metacard workspaceMetacard) {
+    final String host =
+        SystemBaseUrl.getProtocol() + SystemBaseUrl.getHost() + ":" + SystemBaseUrl.getPort();
+    return actionRegistry
+        .list(workspaceMetacard)
+        .stream()
+        .filter(action -> action.getId().startsWith("catalog.data.metacard.list"))
+        .map(
+            action -> {
+              // Work-around for paths being behind VPCs with non-public DNS values
+              final String url = action.getUrl().toString().replaceFirst(host, "");
```

Comment:
❓ Do we have to do this for other actions as well?

Reply:
I am not sure, however I think this can be removed after #3225 is merged.